app/workers/concerns/rate_limit_dump_worker.rb

```diff
@@ -12,8 +14,25 @@ module RateLimitDumpWorker
           "#{resource_type}_#{resource_id}_#{medium_id}_data_dump_worker"
         },
         enabled: ->(_resource_id, _resource_type, _medium_id, requester_id=nil) {
-          requester_id.blank? || !User.find(requester_id).admin?
+          congestion_enabled?(requester_id)
         }
       }
   end
+
+  module ClassMethods
+    def congestion_enabled?(requester_id)
+      # if the user is missing, should only happen via the rails console
+      return false if requester_id.blank?
+
+      if (skip_id_env = ENV.fetch('SKIP_DUMP_RATE_LIMIT_USER_IDS', nil))
+        skip_ids = skip_id_env.split(',').map(&:to_i)
+        skip_ids.include?(requester_id)
+        # false if the user is a special skip rate limit user
+        false
```

Comment:
I don't quite follow how this part works, maybe just because Ruby's implicit returns confuse me. Does execution only get as far as line 31 if line 29 evaluates to `true`? How does this return `true` if the user isn't a special user?

Reply:
This looked like a bug, i've modified to simplify the logic by using initializers to extract the skip user ids